### DISCUSSION TITLE
Upgrading from version 3.7.0 -> 4.5.2

### DISCUSSION THREAD

**Discussion by KevinHobbsSeattle** [nextgenhealthcare/connect#6318] [Frequently Asked Questions] [answered]
Is there any issues or things to be aware of when upgrading from very old version (3.7.0) to latest version (4.5.2) ?
Is this advisable or should there be multiple steps. For example is there any database migrations etc that would need to be performed along the way, upgrading to key versions as an interim step along the way to 4.5.2?
Any other gotchas we should be away of. Are there any breaking changes to the API along the way.

**Comment by pacmano1** [comment]
Read the release notes for each release between 3.7.1 and 4.5.2.    The changes are documented.

**Reply by pacmano1** [reply]
In general, if you are not using custom lib or custom java libraries, upgrades go well.   You still need to consider the release notes though.   Please read the manual on java 9+ required options if using java 9 or later.

**Reply by KevinHobbsSeattle** [reply]
We're running openjdk version "11.0.23" with Mirth 3.7.0 and will be upgrading to, hmm not sure, I guess at least openjdk version "17.0.12", or maybe later. I just finished reading all the release notes and what's new for 3.7.1 -> 4.5.2 and there looks like a few things to take note. Some TLS protocol we will want to verify with outside vendors, possibly some log4j property changes, the DELETE method in the API got changed for patient ID param, SFTP ssh-rsa disabled by default, but no major db changes that would cause us to have to upgrade to an intermediate version first. Nice job Mirth Connect!

Your comment about "Please read the manual on java 9+ required options if using java 9 or later." Would you be able to expand further on that? I'm assuming it's standard things with the language features that we may have to upgrade in our custom java library that we load as a resource? Also, there's probably some deprecated java things to manage.

Thanks again for the response.

**Reply by pacmano1** [reply]
No, it's the required options in mcserver.vmoptions and mcservice.vmoptions for java9+.

https://docs.nextgen.com/bundle/Mirth_User_Guide_4_5_0/page/connect/connect/topics/c_Using_Java_9_or_greater_mirth_connect_ug.html

**Reply by pacmano1** [reply, marked as answer]
Also - best to move your custom jars to resource directories vs custom lib.   And read https://docs.nextgen.com/bundle/Mirth_User_Guide_4_5_0/page/connect/connect/topics/c_Resources_Table_Columns_resource_set_connect_ug.html on the classloader stuff.
